### ISSUE TITLE
Language Code missing in lang_codes.json can cause Admin Console init to fail

### ISSUE THREAD

**Issue by holta** [iiab/iiab#2993]
How can we help @rickgurmet work through this failure to access Admin Console?

Output of `sudo iiab-diagnostics` is here: http://sprunge.us/rI2zR9?en

![image](https://user-images.githubusercontent.com/2458907/135568052-449e5c05-0843-4c33-8ecc-b8eb4d84bda1.png)

**Comment by tim-moody** [comment]
reboot and try again. if still fails to initialize:

1. ssh into server as root
2. systemctl stop iiab-cmdsrv
3. cd /opt/admin/cmdsrv
4. ./iiab-cmdsrv3.py
5. try again and note any errors

**Comment by rickgurmet** [comment]
I did all the steps you mention but after step 4, there is a line that says "in job_minder_thread" and nothing happens after that, just stays there. 
Here is the sprunge.us URL, http://sprunge.us/rgkho1

**Comment by tim-moody** [comment]
and you reran admin console?

**Comment by tim-moody** [comment]
"in job_minder_thread" is waiting for a command

**Comment by tim-moody** [comment]
This was in js console

jquery.min.js:2 jQuery.Deferred exception: Cannot read properties of undefined (reading 'locname') TypeError: Cannot read properties of undefined (reading 'locname')
    at String.<anonymous> (http://10.8.0.38/admin/js/zim_functions.js:139:39)
    at Function.each (http://10.8.0.38/common/js/jquery.min.js:2:3003)
    at procZimGroups (http://10.8.0.38/admin/js/zim_functions.js:135:5)
    at procZimCatalog (http://10.8.0.38/admin/js/zim_functions.js:409:3)
    at Array.initDone (http://10.8.0.38/admin/js/admin_console.js:2307:5)
    at e (http://10.8.0.38/common/js/jquery.min.js:2:30038)
    at t (http://10.8.0.38/common/js/jquery.min.js:2:30340) undefined

lbj is missing from lang_codes.json

**Comment by holta** [comment]
Should we close this now, or keep it as a placeholder for other languages that also need fixing?

**Comment by holta** [comment]
Here are 36 other missing languages that @tim-moody mentioned:

(that might also need to be addressed in future)

```
Language code lbj not in langCodes.
admin_console.js:2156 Language code ban not in langCodes.
admin_console.js:2156 Language code lfn not in langCodes.
admin_console.js:2156 Language code din not in langCodes.
admin_console.js:2156 Language code gor not in langCodes.
admin_console.js:2156 Language code nai not in langCodes.
admin_console.js:2156 Language code sgs not in langCodes.
admin_console.js:2156 Language code grc not in langCodes.
admin_console.js:2156 Language code rup not in langCodes.
admin_console.js:2156 Language code azb not in langCodes.
admin_console.js:2156 Language code nan not in langCodes.
admin_console.js:2156 Language code arp not in langCodes.
admin_console.js:2156 Language code rmr not in langCodes.
admin_console.js:2156 Language code cbk not in langCodes.
admin_console.js:2156 Language code kbp not in langCodes.
admin_console.js:2156 Language code ady not in langCodes.
admin_console.js:2156 Language code brx not in langCodes.
admin_console.js:2156 Language code crh-latn not in langCodes.
admin_console.js:2156 Language code lzh not in langCodes.
admin_console.js:2156 Language code be-tarask not in langCodes.
admin_console.js:2156 Language code inh not in langCodes.
admin_console.js:2156 Language code vro not in langCodes.
admin_console.js:2156 Language code sat not in langCodes.
admin_console.js:2156 Language code hbs not in langCodes.
admin_console.js:2156 Language code atj not in langCodes.
admin_console.js:2156 Language code lrc not in langCodes.
admin_console.js:2156 Language code jam not in langCodes.
admin_console.js:2156 Language code myn not in langCodes.
admin_console.js:2156 Language code oji not in langCodes.
admin_console.js:2156 Language code enm not in langCodes.
admin_console.js:2156 Language code bgs not in langCodes.
admin_console.js:2156 Language code olo not in langCodes.
admin_console.js:2156 Language code kld not in langCodes.
admin_console.js:2156 Language code ale not in langCodes.
admin_console.js:2156 Language code yue not in langCodes.
admin_console.js:2156 Language code shn not in langCodes.
admin_console.js:2156 Language code mnw not in langCodes.

**Comment by holta** [comment]
Here are 36 other missing languages that @tim-moody mentioned:

(that might also need to be addressed in future)

```
Language code lbj not in langCodes.
admin_console.js:2156 Language code ban not in langCodes.
admin_console.js:2156 Language code lfn not in langCodes.
admin_console.js:2156 Language code din not in langCodes.
admin_console.js:2156 Language code gor not in langCodes.
admin_console.js:2156 Language code nai not in langCodes.
admin_console.js:2156 Language code sgs not in langCodes.
admin_console.js:2156 Language code grc not in langCodes.
admin_console.js:2156 Language code rup not in langCodes.
admin_console.js:2156 Language code azb not in langCodes.
admin_console.js:2156 Language code nan not in langCodes.
admin_console.js:2156 Language code arp not in langCodes.
admin_console.js:2156 Language code rmr not in langCodes.
admin_console.js:2156 Language code cbk not in langCodes.
admin_console.js:2156 Language code kbp not in langCodes.
admin_console.js:2156 Language code ady not in langCodes.
admin_console.js:2156 Language code brx not in langCodes.
admin_console.js:2156 Language code crh-latn not in langCodes.
admin_console.js:2156 Language code lzh not in langCodes.
admin_console.js:2156 Language code be-tarask not in langCodes.
admin_console.js:2156 Language code inh not in langCodes.
admin_console.js:2156 Language code vro not in langCodes.
admin_console.js:2156 Language code sat not in langCodes.
admin_console.js:2156 Language code hbs not in langCodes.
admin_console.js:2156 Language code atj not in langCodes.
admin_console.js:2156 Language code lrc not in langCodes.
admin_console.js:2156 Language code jam not in langCodes.
admin_console.js:2156 Language code myn not in langCodes.
admin_console.js:2156 Language code oji not in langCodes.
admin_console.js:2156 Language code enm not in langCodes.
admin_console.js:2156 Language code bgs not in langCodes.
admin_console.js:2156 Language code olo not in langCodes.
admin_console.js:2156 Language code kld not in langCodes.
admin_console.js:2156 Language code ale not in langCodes.
admin_console.js:2156 Language code yue not in langCodes.
admin_console.js:2156 Language code shn not in langCodes.
admin_console.js:2156 Language code mnw not in langCodes.

**Comment by holta** [comment]
Also @m-anish will likely benefit from this Ladakhi-related language, if this might be added as well:

> zau – Zangskari

**Comment by jvonau** [comment]
status of this issue is?

**Comment by holta** [comment]
Thankfully the narrow issue of the Ladakhi language (lbj) is now resolved:

- PR #2994

Which brings us up to 310 languages in:

- https://github.com/iiab/iiab/blob/master/roles/www_base/files/html/assets/lang_codes.json

The deeper question is how do we actively welcome more languages &mdash; by promoting this ticket or others ?
